gui/modlistman.lua

```diff
@@ -0,0 +1,446 @@
+            key = "SELECT",
+            label = "Load",
+            on_activate = function()
+                local idx, current = presetList:getSelected()
+
+                if not idx then
+                    return
+                end
+
+                load_preset(idx)
+            end,
+        },
+
+        widgets.HotkeyLabel{
+            frame = { l = 17, b = 2, w = 15 },
+            key = "CUSTOM_D",
+            label = "Delete",
+            on_activate = function()
+                local idx, current = presetList:getSelected()
+
+                if not idx then
+                    return
+                end
+
+                dialogs.showYesNoPrompt(
+                    "Confirmation",
+                    "Delete " .. current.text .. "?",
+                    nil,
+                    function()
+                        remove_preset(idx)
+                        refresh_list()
+                    end
+                )
+            end,
+        },
+
+        widgets.HotkeyLabel{
+            frame = { l = 1, b = 0, w = 20 },
+            key = "CUSTOM_Q",
+            label = "Set as default",
+            on_activate = function()
+                local idx, current = presetList:getSelected()
+
+                if not idx then
+                    return
+                end
+
+                toggle_default(idx)
+                refresh_list()
+            end
+        },
+    }
+end
+
+ModmanageScreen = defclass(ModmanageScreen, gui.ZScreen)
+ModmanageScreen.ATTRS {
+    focus_path = "modman_screen",
+    defocusable = false,
+}
+
+function ModmanageScreen:init()
+    self:addviews{
+        ModmanageMenu{}
+    }
+end
+
+ModmanageOverlay = defclass(ModmanageOverlay, overlay.OverlayWidget)
+ModmanageOverlay.ATTRS {
+    frame = { w=30, h=1 },
+    default_pos = {x=5, y=-5},
+    viewscreens = {"new_region"},
+    default_enabled=true,
+    overlay_only=true,
+    hotspot=true,
+    overlay_onupdate_max_freq_seconds=0,
+}
+
+function ModmanageOverlay:init()
+    self:addviews{
+        widgets.HotkeyLabel{
+            frame = {l=0, t=0, w = 20, h = 1},
+            key = "CUSTOM_M",
+            view_id = "modman_open",
+            label = "Mod Manager",
+            on_activate = function()
+                ModmanageScreen{}:show()
+            end,
+        },
+    }
+end
+
+function ModmanageOverlay:overlay_onupdate(viewscreen)
+    self.visible = is_modlist_visible()
+
+    for k, v in pairs(self.subviews) do
+        v.visible = self.visible
+    end
+end
+
+OVERLAY_WIDGETS = {
+    button = ModmanageOverlay,
+}
+
+local default_applied = false
+
+dfhack.onStateChange[GLOBAL_KEY] = function(sc)
+    if sc == SC_VIEWSCREEN_CHANGED then
+        local vs = get_newregion_viewscreen()
+        if vs and not default_applied then
+            default_applied = true
+            for i, v in ipairs(presets_file.data) do
+                if v.default then
+                    load_preset(i)
```

Comment:
it would be nice to inform the player that their default settings have been successfully applied. popping up a message that has to be dismissed would kind of defeat the point of having a convenient default, though. What about if a message appears on the lower left of the screen when the new_region viewscreen first appears (and this code segment takes effect). the message could be cleared when the viewscreen is exited or a subscreen is entered.

Reply:
<img width="889" alt="2023-05-04_16+47+41" src="https://user-images.githubusercontent.com/76485880/236226888-45a9abbf-8456-4b97-bfbe-f5cc3241b77c.png">

bottom-left, stays up for 5 seconds, is a separate overlay
included in latest commit

Reply:
perfect. Thanks!